google-cloud-storage/src/main/java/com/google/cloud/storage/GapicUnbufferedReadableByteChannel.java

```diff
@@ -237,12 +237,26 @@ private final class LazyServerStreamIterator implements Iterator<ReadObjectRespo
 
     @Override
     public boolean hasNext() {
-      return ensureResponseIteratorOpen().hasNext();
+      try {
+        return ensureResponseIteratorOpen().hasNext();
+      } catch (RuntimeException e) {
+        if (!result.isDone()) {
+          result.setException(StorageException.coalesce(e));
```

Comment:
why do we set the exception on the result only if the result is not complete? What case is this covering?

Reply:
A call to `hasNext()`/`next()` could error any time. More often than not, as long as the object exists, this code will not run, but when the object doesn't exist this ensures we communicate that to the result as well as any call to `read`